Add imx8qm Display VM configuration

ruizjuanpablo opened the pull request:
Added a configuration file to build the imx8qm image with the requiremnts to launch the Display VM with GPU/HDMI passthrough. This configuration file adds the imx8qm_display_vm.nix overlay file which will have all the required overlays.

In this commit the overlay to support the GPU/HDMI passthrough from u-boot was also added.

Signed-off-by: Juan Pablo Ruiz <juanpablo.ruiz@tii.ae>

unbel13ver commented on overlays/bsp/u-boot/imx8qm/patches/0001-Add-support-for-Display-VM.patch:
```diff
@@ -0,0 +1,64 @@
+ 	"boot_fdt=try\0" \
+-	"fdtfile=imx8qm-mek-hdmi.dtb\0" \
++	"fdtfile=imx8qm-mek-kvm-host.dtb\0" \
++	"fdt_file=imx8qm-mek-kvm-host.dtb\0" \
```
Can you please clarify the idea of having two different names for single value?

unbel13ver replied:
Oh, and also I think you need to change fdt file in `overlays/imx8qm.nix` file

ruizjuanpablo replied:
Thanks for noting this. I will review and keep only the fdt file that is needed. 

For the overlays/imx8qm.nix you are right, I need to update this with the new fdt file. Let me do this when I integrate the Linux kernel patches that build this fdt file.

unbel13ver replied:
Sorry, I do not completely understand how this .dtb file is installed to the boot partition. Currently it is done here:
https://github.com/tiiuae/build-configurations/blob/01a40cc360ab7b8d8d6a2a0bae21ecceff53a326/overlays/imx8qm.nix#L46

ruizjuanpablo replied:
It is not right now, but it will be when I integrate the Linux kernel patches.

unbel13ver commented on overlays/imx8qm_display_vm.nix:
```diff
@@ -0,0 +1,9 @@
+  
+  ubootImx8 = prev.ubootImx8.overrideAttrs (old: {
+    patches = old.patches ++ [
+      ./bsp/u-boot/imx8qm/patches/0001-Add-support-for-Display-VM.patch
```
Let's put this patch to the imx-uboot.nix file directly instead?

ruizjuanpablo replied:
Due this patch should be applied only for the imx8qm-display-vm-config.nix configuration, it is necessary that this patch stays in an overlay.

unbel13ver replied:
Yes, now I see, thanks!

unbel13ver commented:
There is a typo in commit message also: `requiremnts` - could you please fix that also?

ruizjuanpablo commented:
Thanks, the changes will be submitted.

Commit:
Add imx8qm Display VM configuration

Added a configuration file to build the imx8qm image with the
requirements to launch the Display VM with GPU/HDMI passthrough.
This configuration file adds the imx8qm_display_vm.nix overlay file
which will have all the required overlays.

In this commit the overlay to support the GPU/HDMI passthrough from
u-boot was also added.

Signed-off-by: Juan Pablo Ruiz <juanpablo.ruiz@tii.ae>
